server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -0,0 +1,394 @@
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            ((request, channel, task) -> channel.sendResponse(handleExtensionRequest(request)))
+        );
+    }
+
+    @Override
+    public PluginsAndModules info() {
+        return null;
+    }
+
+    /*
+     * Load and populate all extensions
+     */
+    private void discover() throws IOException {
+        logger.info("Extensions Config Directory :" + extensionsPath.toString());
+        if (!FileSystemUtils.isAccessibleDirectory(extensionsPath, logger)) {
+            return;
+        }
+
+        List<Extension> extensions = new ArrayList<Extension>();
+        if (Files.exists(extensionsPath.resolve("extensions.yml"))) {
+            try {
+                extensions = readFromExtensionsYml(extensionsPath.resolve("extensions.yml")).getExtensions();
+            } catch (IOException e) {
+                throw new IOException("Could not read from extensions.yml", e);
+            }
+            for (Extension extension : extensions) {
+                loadExtension(extension);
+            }
+            if (!uninitializedExtensions.isEmpty()) {
+                logger.info("Loaded all extensions");
+            }
+        } else {
+            logger.info("Extensions.yml file is not present.  No extensions will be loaded.");
+        }
+    }
+    
+    /**
+     * Loads a single extension
+     * @param extension The extension to be loaded
+     */
+    private void loadExtension(Extension extension) throws IOException {
+        try {
+            uninitializedExtensions.add(
+                new DiscoveryExtensionNode(
+                    extension.getName(),
+                    extension.getUniqueId(),
+                    // placeholder for ephemeral id, will change with POC discovery
+                    extension.getUniqueId(),
+                    extension.getHostName(),
+                    extension.getHostAddress(),
+                    new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
+                    new HashMap<String, String>(),
+                    Version.fromString(extension.getOpensearchVersion()),
+                    new PluginInfo(
+                        extension.getName(),
+                        extension.getDescription(),
+                        extension.getVersion(),
+                        Version.fromString(extension.getOpensearchVersion()),
+                        extension.getJavaVersion(),
+                        extension.getClassName(),
+                        new ArrayList<String>(),
+                        Boolean.parseBoolean(extension.hasNativeController())
+                    )
+                )
+            );
+            logger.info("Loaded extension: " + extension);
+        } catch (IllegalArgumentException e) {
+            logger.error(e.toString());
```

Comment:
Shouldn't we fail with `throw new IllegalStateException()` instead of swallowing extension initialization issues?